tests/typing/callable.py

```diff
@@ -55,20 +67,23 @@ def create(cls) -> Animal:
 
 
 async def _async9() -> None:
-    animal1: Animal = await provider9(1, 2, 3, b="1", c=2, e=0.0)  # type: ignore
-    animal2: Animal = await provider9.async_(1, 2, 3, b="1", c=2, e=0.0)
+    await provider9(1, 2, 3, b="1", c=2, e=0.0)  # type: ignore[misc]
+    cat9 = await provider9.async_(1, 2, 3, b="1", c=2, e=0.0)
+    assert_type(cat9, Cat)
 
 
 # Test 10: to check the .provides
 provider10 = providers.Callable(Cat)
-provides10: Optional[Callable[..., Cat]] = provider10.provides
-assert provides10 is Cat
+provides10 = provider10.provides
+assert_type(provides10, Optional[Callable[..., Cat]])
 
 # Test 11: to check the .provides for explicit typevar
 provider11 = providers.Callable[Animal](Cat)
-provides11: Optional[Callable[..., Animal]] = provider11.provides
-assert provides11 is Cat
+provides11 = provider11.provides
+assert_type(provides11, Optional[Callable[..., Animal]])
+
 
 # Test 12: to check string imports
-provider12: providers.Callable[Dict[Any, Any]] = providers.Callable("builtins.dict")
+# TODO: Use T_Any as Callable typevar? Then remove type-ignore
```

Comment:
Can't infer typevar without explicit typing, mypy raises error here, but we can change it to typevar with default Any to fix it